### PR TITLE
feat/full-text-search

### PR DESCRIPTION
**Pull request by 0xori**
task search by title or description

### CODE REVIEW

**Comment by rootux** [comment]
Looking amazing.

@0xori Did you checked how that works for a really big projects? 3000 tasks one?

**Comment by rootux** [comment]
We need to create a project with 2000 tasks and see how that works there

**Comment by 0xori** [comment]
@rootux I didn't checked. Yes we definitely need to do this

File: src/views/pages/tasks/tasks-page.js [hunk 1]
```diff
@@ -420,12 +455,14 @@ export class TasksPage extends Component {
     return (
       <div>
         <div className="g-col">
-          { <TaskFilters
-            filter = { this.props.filterType }
-            selectedProject = { this.props.selectedProject }
-            projectUrl = { projectUrl } //TODO - should be from state
-            labels = { this.state.labelPool }
-            onLabelChange = { this.onLabelChanged }
+          {<TaskFilters
```
Comment (netanelben) [inline src/views/pages/tasks/tasks-page.js:465]: No need to wrap it with `{}` unless there is a render-condition for instance: `{isVisible && <Component/>}`

**Comment by netanelben** [comment]
Can we combine the two search boxes into one ?

Screen Shot 2019-10-12 at 12 09 48

File: src/tasks/selectors.js [hunk 2]
```diff
@@ -62,5 +63,16 @@ export const taskFilters = {
         return ((task.assignee && (task.assignee.id === auth)) ||
           (task.creator && task.creator.id === auth));
       });
-  }
+  },
+
+  query: (tasks, filter) => {
```
Comment (netanelben) [inline src/tasks/selectors.js:68]: `filter.text` already init as empty string, so the check is redundant.
`filter` function will return the `tasks` array if nothing found so this selector can be:

```JS
query: (tasks, filter) =>
    tasks.filter(task => {
      const filterText = filter.text.toLowerCase();
      const taskTitle = task.title.toLowerCase();
      const taskDescription = task.description.toLowerCase();

      return taskTitle.includes(filterText) || taskDescription.includes(filterText);
})
```

**Comment by rootux** [comment]
@0xori I've created a project with large size of tasks **(1936 tasks)** to do some stress testing - project is `kjhkjhjkhkj`-
http://localhost:3000/kjhkjhjkhkj/task/1?complete=false

It seems to work great. so we are loading in memory those tasks - but simply don't display them which now makes a lot of sense for this Frontend searching. Amazing Work!

@netanelben Thanks for you input - super valid - love your coding style. As to design @0xori suggested we have a design session go on it - Frankly I don't think we need two search boxes - if that single one can simply filter both - that would be amazing - the only issue I can see is if someone want to filter ONLY by a tag - for example a certain camp has a tag "My-Camp" and upon press he doesn't want to search for "My-Camp" but to filter using this tag.

**Comment by rootux** [comment]
Screen Shot 2019-10-12 at 17 04 20

Some really cool (hidden?) feature - It uses you to do all the right AND for the filters - 
you can say task with query "magic" and of type "My tasks" OR
Task with query "magic" and type "Camps"

This is great!
So probably we need a better UX to combine the two searches

File: src/tasks/selectors.js (hunk 2 above)
Comment (rootux) [reply src/tasks/selectors.js:68]: Actually filter.text is failing when I remove the check for null

**Comment by rootux** [comment]
@netanelben Something I can't fix - Try to search for a word that doesn't exist - "This is some random text" - then press on the button in the image 
Screen Shot 2019-10-12 at 19 47 44

I want the query text to clear itself - Not sure how to do so. any ideas?

**Comment by rootux** [comment]
Merging for now. Cause it's working pretty well.

**Comment by 0xori** [comment]
>Frankly I don't think we need two search boxes - if that single one can simply filter both - that would be amazing - the only issue I can see is if someone want to filter ONLY by a tag - for example a certain camp has a tag "My-Camp" and upon press he doesn't want to search for "My-Camp" but to filter using this tag.

@rootux 
I think we should treat tags like the way trello is filtering by labels.

**Comment by netanelben** [comment]
> @netanelben Something I can't fix - Try to search for a word that doesn't exist - "This is some random text" - then press on the button in the image
> Screen Shot 2019-10-12 at 19 47 44
> 
> I want the query text to clear itself - Not sure how to do so. any ideas?

Im checking that now :)

**Pull request merged**